File: Data Structures & Algorithms/daily-temperatures/submission-1.py
class Solution:
    def dailyTemperatures(self, temperatures: List[int]) -> List[int]:
        res = [0] * len(temperatures)
        stack = []

        for index, temperature in enumerate(temperatures):
            while stack and temperature > stack[-1][0]:
                print(stack)
                stackTemperature, stackIndex = stack.pop()
                res[stackIndex] = index - stackIndex
            stack.append((temperature, index))
            print(res)
        return res

File: Data Structures & Algorithms/validate-parentheses/submission-1.js
class Solution {
    /**
     * @param {string} s
     * @return {boolean}
     */
    isValid(s) {
        let stack = [];
        for(let i = 0; i < s.length; i++)
        {
            if(s[i] == '[' || s[i] == '{' || s[i] == '(')
            {
                stack.push(s[i])
            }
            else
            {
                let lastEl = stack.pop();
                let combination = lastEl + s[i];
                console.log(stack);
                console.log(combination);
                if(combination != '{}' && combination != '[]' && combination != '()')
                {
                    return false;
                }
            }
            
        }

        if(stack.length != 0) return false;
        return true;
    }
}
